core/src/main/java/io/grpc/DnsNameResolver.java

```diff
@@ -161,7 +163,8 @@ public void run() {
             servers.add(
                 new ResolvedServerInfo(new InetSocketAddress(inetAddr, port), Attributes.EMPTY));
           }
-          savedListener.onUpdate(servers, Attributes.EMPTY);
+          savedListener.onUpdate(
+              Collections.<List<ResolvedServerInfo>>singletonList(servers),Attributes.EMPTY);
```

Comment:
I can solve the casting issue by assigning the array list to a List instead of an ArrayList a few lines higher.  I'll just go that route.